app/controllers/executions_controller.rb

```diff
@@ -11,22 +11,26 @@ def results
       return
     end
 
-    executions = Execution.where(submission: submission)
-    failed_test_ids = executions
-                      .reject(&:passed?)
-                      .map(&:test_id)
+    total_tests = submission.problem.tests.count
 
-    num_passed = executions.size - failed_test_ids.size
+    executions = Execution.where(submission: submission)
+    executions_passed = executions.select(&:passed?)
+    executions_failed = executions.select(&:failed?)
+    executions_errored = executions.select(&:errored?)
 
-    failed_test_hints = []
-    unless failed_test_ids.empty?
-      failed_test_hints = Test.find(failed_test_ids).pluck(:hint)
-    end
+    failed_test_hints = Test.find(executions_failed.map(&:test_id)).pluck(:hint)
 
     render status: :ok, json: {
-      total_tests: executions.size,
-      num_passed: num_passed,
-      failed_test_hints: failed_test_hints
+      executions: {
+        passed: executions_passed,
```

Comment:
If the execution itself contains whether it passed/failed/errored, we should try to stick to our schema unless we have reason not to. Unless I'm misunderstanding the `.select(&:passed?)`

Reply:
i.e. I dont think separating out the executions gives the frontend more data unless Im wrong